src/app/@ansyn/plugins/openlayers/open-layers-north-calculations/plugin/north-calculations-plugin.ts

```diff
@@ -61,33 +62,25 @@ export class NorthCalculationsPlugin extends BaseImageryPlugin {
 	}
 
 	getCorrectedNorth(): Observable<INorthData> {
-		const mapObject = this.ActiveMap.mapObject;
-		const size = mapObject.getSize();
-		const olCenterView = mapObject.getCoordinateFromPixel([size[0] / 2, size[1] / 2]);
-		const olCenterViewWithOffset = mapObject.getCoordinateFromPixel([size[0] / 2, (size[1] / 2) - 1]);
-
-
-		if (!olCenterView) {
-			return Observable.throw('no coordinate for pixel');
-		}
-
-		return this.projectPoints([olCenterView, olCenterViewWithOffset]).map((projectedPoints: Point[]) => {
+		return Observable.create((observer: Observer<any>) => {
```

Comment:
I think that a clearer approach may be declaring `getCenters(): Observable<ol.Coordinate[]>` and then chaining it to `projectPoint()` via the `map` operator.